src/services/EdFi.DataManagementService.Api/Core/Validation/DocumentValidator.cs

```diff
@@ -22,15 +51,59 @@ public interface IDocumentValidator
 
 public class DocumentValidator(ISchemaValidator schemaValidator) : IDocumentValidator
 {
-    public IEnumerable<string>? Validate(JsonNode? documentBody, ValidatorContext validatorContext)
+    public IEnumerable<string>? ValidateJsonFormat(JsonNode? documentBody)
     {
         var formatValidationResult = documentBody.ValidateJsonFormat();
 
         if (formatValidationResult != null && formatValidationResult.Any())
         {
             return formatValidationResult;
         }
+        return null;
+    }
+
+    public EvaluationResults Evaluate(JsonNode? documentBody, ValidatorContext validatorContext)
+    {
+        EvaluationOptions? validatorEvaluationOptions =
+            new() { OutputFormat = OutputFormat.List, RequireFormatValidation = true };
+
+        var resourceSchemaValidator = schemaValidator.GetSchema(validatorContext);
+        return resourceSchemaValidator.Evaluate(documentBody, validatorEvaluationOptions);
+    }
+
+    public bool PruneOverPostedData(
+        JsonNode? documentBody,
+        EvaluationResults evaluationResults,
+        out JsonNode? prunedDocumentBody
+    )
+    {
+        prunedDocumentBody = documentBody;
+
+        if (documentBody == null)
+            return false;
+
+        var additionalProperties = evaluationResults
+            .Details.Where(r =>
+                r.EvaluationPath.Segments.Any() && r.EvaluationPath.Segments[^1] == "additionalProperties"
+            )
+            .ToList();
+
+        if (additionalProperties.Count == 0)
+            return false;
+
+        foreach (var additionalProperty in additionalProperties)
+        {
+            JsonObject jsonObject = documentBody.AsObject();
+            var prunedJsonObject = jsonObject.RemoveProperty(additionalProperty.InstanceLocation.Segments);
+            documentBody = JsonNode.Parse(prunedJsonObject.ToJsonString())!;
```

Comment:
Agree `Parse` should never be null here, but suggest assert like "prunedJsonObject deserialization failure".